demisto_sdk/commands/init/tests/initiator_test.py

```diff
@@ -504,3 +560,51 @@ def test_script_init(monkeypatch, initiator, tmpdir):
     assert os.path.isdir(script_path)
     assert {f"{SCRIPT_NAME}.py", f"{SCRIPT_NAME}.yml", f"{SCRIPT_NAME}_test.py",
             "README.md", "test_data"} == script_dir_files
+
+
+def test_script_init_with_ignore_secrets(initiator, tmpdir, monkeypatch, mocker):
+    """
+    Tests `script_init` function with a given script template name.
+    Given
+        - A script template - BaseScript.
+    When
+        - Running the init command with secrets ignore.
+    Then
+        - Ensure the function's ('script_init') return value is True
+        - Ensure script directory with the desired script name is created successfully.
+        - Ensure script directory contain all files.
+        - Ensure .secrets-ignore file is not empty.
+    """
+    monkeypatch.setattr('builtins.input', generate_multiple_inputs(deque(['6.0.0', 'y'])))
+    white_list = {
+        "files": [],
+        "iocs": {},
+        "generic_strings": []
+    }
+    temp_test_dir = os.path.join(tmpdir, 'Tests')
+    os.makedirs(temp_test_dir, exist_ok=True)
+    with open(f'{temp_test_dir}/secrets_white_list.json', 'w') as f:
+        json.dump(white_list, f)
+
+    temp_pack_dir = os.path.join(tmpdir, 'Packs', PACK_NAME)
+    os.makedirs(temp_pack_dir, exist_ok=True)
+    secrets_ignore_path = f'{temp_pack_dir}/.secrets-ignore'
+
+    initiator.output = temp_pack_dir
+    initiator.dir_name = SCRIPT_NAME
+    initiator.template = DEFAULT_SCRIPT
+    initiator.category = 'Utilities'
+
+    script_path = os.path.join(temp_pack_dir, SCRIPT_NAME)
+    mocker.patch('demisto_sdk.commands.init.initiator.get_pack_name', return_value='PackName')
+
+    with ChangeCWD(tmpdir):
+        res = initiator.script_init()
+
+    script_dir_files = {file for file in listdir(script_path)}
+
+    assert res
+    assert os.path.isdir(script_path)
+    assert {f"{SCRIPT_NAME}.py", f"{SCRIPT_NAME}.yml", f"{SCRIPT_NAME}_test.py",
+            "README.md", "test_data"} == script_dir_files
+    assert os.stat(secrets_ignore_path).st_size > 0
```

Comment:
Why aren't we testing here the files diff like in the integration test?

Reply:
Lets try to be consistent with the UT.
If you have a test for a script that is checking the same as the integration it should look the same.
Also, please pay attention when copying code from others, this should be done carefully and with a lot of attention - we should fully understand what was done there and ask ourselves whether its relevant for our case